src/index.js

```diff
@@ -520,6 +553,206 @@ class EventsCoreAPI {
    * =========================================================================
    */
 
+  /**
+   * Wrapper to check the event received by webhook
+   * and return decoded (if encoded) and properly parsed payload
+   * @param {*} event event payload received by webhook
+   * @returns decoded and properly parsed payload
+   */
+  __getProperPayload (event) {
+    let decodedJsonPayload
+    try {
+      if (this.__isBase64Encoded(event)) {
+        decodedJsonPayload = JSON.parse(Buffer.from(event, 'base64').toString('utf-8'))
+      } else {
+        // parsing for non-encoded json payloads (e.g. custom events)
+        decodedJsonPayload = JSON.parse(event)
+      }
+    } catch (error) {
+      logger.error('error occured while checking payload' + error.message)
+      return this.__genErrorResponse(400, 'Failed to understand the payload')
+    }
+    return decodedJsonPayload
+  }
+
+  /**
+   * Checks for the payload type if it is base64 encode or not
+   * 
+   * @param {*} eventPayload - the event payload received by the consumer webhook
+   * @returns {boolean} true if encoded or false
+   */
+  __isBase64Encoded (eventPayload) {
```

Comment:
This method can go in the generic `helpers.js` file.

Reply:
okay

Reply:
moved it to a signature utils new file